src/Support/EagerLoader.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Thunk\Verbs\Support;
+
+use Arr;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Collection;
+use ReflectionClass;
+use ReflectionProperty;
+use Thunk\Verbs\Attributes\Projection\EagerLoad;
+use Thunk\Verbs\Event;
+
+class EagerLoader
+{
+    public static function load(Event ...$events)
+    {
+        return (new static($events))();
+    }
+
+    public function __construct(
+        /** @var Event[] */
+        protected array $events,
+    ) {}
+
+    public function __invoke()
+    {
+        $discovered = collect($this->events)
+            ->map($this->discover(...))
+            ->reduce(function (array $map, Collection $discovered) {
+                foreach ($discovered as [$class_name, $event, $id_property, $target_property]) {
+                    $map['load'][$class_name][] = $event->{$id_property};
+                    $map['fill'][$class_name][$event->{$id_property}][] = [$event, $target_property];
+                }
+
+                return $map;
+            }, ['load' => [], 'fill' => []]);
+
+        /** @var class-string<Model> $class_name */
+        foreach ($discovered['load'] as $class_name => $keys) {
+            $class_name::query()
+                ->whereIn((new $class_name)->getKeyName(), $keys)
+                ->eachById(function (Model $model) use ($discovered) {
+                    foreach ($discovered['fill'][$model::class][$model->getKey()] as [$event, $target_property]) {
+                        // This let's us set the property even if it's protected
+                        (fn () => $this->{$target_property} = clone $model)(...)->call($event);
```

Comment:
I'm not certain this should be `clone` here — there are pros and cons to both approaches. Deserves discussion.

Reply:
Namely: with `clone` you avoid subtle bugs around recursion and side effects (although some of these are addressed in the latest Laravel 11 release). Without `clone`, each event will be mutating the same model, which is probably preferable. Otherwise changes in one even won't impact others correctly.

Reply:
Decision: don't clone.